### PR TITLE
Fix windows with BorderOnly decoration being unresizable

### LINKED ISSUE #6786: Window with SystemDecorations set to "BorderOnly" cannot be resized
**Describe the bug**
If you have a Window with SystemDecorations="BorderOnly" and CanResize="True", there is no way to grab and resize a window. I'm not sure there's a real difference between "BorderOnly" and "None".

**To Reproduce**

1. Create a new Avalonia project from the template Avalonia.NETCoreMVVMApp (I did this in Rider)
2. On the MainWindow set SystemDecorations="BorderOnly" and CanResize="True"
3. Run the application
4. UNEXPECTED BEHAVIOR: cannot resize the window by dragging from the edges or corners.

**Expected behavior**
A window with SystemDecorations="BorderOnly" and CanResize="True" should be resiable.

**Desktop (please complete the following information):**

- OS: Windows 10
- Version: 0.10.8

**Additional context**
I am using the fluent theme, same problem for both light and dark versions.

### PR DESCRIPTION
**Pull request by emmauss**
## What does the pull request do?
Removes the check for decoration in the window message handler for `WM_NCCALCSIZE`. This enables borders to be handled by the default window message handler, thus making resizing possible in BorderOnly and None decoration modes.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues
Fixes #6786

### CODE REVIEW

File: src/Windows/Avalonia.Win32/WindowImpl.AppWndProc.cs [hunk 1]
```diff
@@ -59,7 +59,7 @@ protected virtual unsafe IntPtr AppWndProc(IntPtr hWnd, uint msg, IntPtr wParam,
 
                 case WindowsMessage.WM_NCCALCSIZE:
                     {
-                        if (ToInt32(wParam) == 1 && !HasFullDecorations || _isClientAreaExtended)
+                        if (ToInt32(wParam) == 1 && _isClientAreaExtended)
```
[the remark this answers is not in the document]
Comment (emmauss) [reply src/Windows/Avalonia.Win32/WindowImpl.AppWndProc.cs]: Done.

**Review by kekekeks — APPROVED** [review]

**Comment by cirrusone** [comment]
This fix isn't working for me on v11.0.4 on WSL Ubuntu. Does this fix apply to v11.0.4 or is it still nightly only?

To reproduce, I just created new project and changed 

**MainWindow.axaml:**

```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="using:AvaloniaApplication1.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:views="clr-namespace:AvaloniaApplication1.Views"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication1.Views.MainWindow"
        Icon="/Assets/avalonia-logo.ico"
        Title="AvaloniaApplication1"
        Background="White"
        ExtendClientAreaChromeHints="NoChrome"
        ExtendClientAreaTitleBarHeightHint="-1"
        ExtendClientAreaToDecorationsHint="True"
        SystemDecorations="BorderOnly">
        <views:MainView />
</Window>
```

**AvaloniaApplication1.csproj:**
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <Nullable>enable</Nullable>
    <LangVersion>latest</LangVersion>
    <AvaloniaUseCompiledBindingsByDefault>true</AvaloniaUseCompiledBindingsByDefault>
  </PropertyGroup>

  
  <ItemGroup>
    <AvaloniaResource Include="Assets\**" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Avalonia" Version="11.0.4" />
    <PackageReference Include="Avalonia.Themes.Fluent" Version="11.0.4" />
    <PackageReference Include="Avalonia.Fonts.Inter" Version="11.0.4" />
    <PackageReference Include="CommunityToolkit.Mvvm" Version="8.2.1" />

    <!--Condition below is needed to remove Avalonia.Diagnostics package from build output in Release configuration.-->
    <PackageReference Condition="'$(Configuration)' == 'Debug'" Include="Avalonia.Diagnostics" Version="11.0.4" />
  </ItemGroup>
</Project>
```

There are no grips to resize the window.  I'm trying to create a customized titlebar so am using `SystemDecorations="BorderOnly"` to remove the auto generated titlebar.

Works fine on Windows, just not WSL Ubuntu.

Untitled
Untitled2

**Comment by emmauss** [comment]
> This fix isn't working for me on v11.0.4 on WSL Ubuntu. Does this fix apply to v11.0.4 or is it still nightly only?
> 
> To reproduce, I just created new project and changed
> 
> **MainWindow.axaml:**
> 
> ```
> <Window xmlns="https://github.com/avaloniaui"
>         xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
>         xmlns:vm="using:AvaloniaApplication1.ViewModels"
>         xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
>         xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
>         xmlns:views="clr-namespace:AvaloniaApplication1.Views"
>         mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
>         x:Class="AvaloniaApplication1.Views.MainWindow"
>         Icon="/Assets/avalonia-logo.ico"
>         Title="AvaloniaApplication1"
>         Background="White"
>         ExtendClientAreaChromeHints="NoChrome"
>         ExtendClientAreaTitleBarHeightHint="-1"
>         ExtendClientAreaToDecorationsHint="True"
>         SystemDecorations="BorderOnly">
>         <views:MainView />
> </Window>
> ```
> 
> **AvaloniaApplication1.csproj:**
> 
> ```
> <Project Sdk="Microsoft.NET.Sdk">
>   <PropertyGroup>
>     <TargetFramework>net8.0</TargetFramework>
>     <Nullable>enable</Nullable>
>     <LangVersion>latest</LangVersion>
>     <AvaloniaUseCompiledBindingsByDefault>true</AvaloniaUseCompiledBindingsByDefault>
>   </PropertyGroup>
> 
>   
>   <ItemGroup>
>     <AvaloniaResource Include="Assets\**" />
>   </ItemGroup>
> 
>   <ItemGroup>
>     <PackageReference Include="Avalonia" Version="11.0.4" />
>     <PackageReference Include="Avalonia.Themes.Fluent" Version="11.0.4" />
>     <PackageReference Include="Avalonia.Fonts.Inter" Version="11.0.4" />
>     <PackageReference Include="CommunityToolkit.Mvvm" Version="8.2.1" />
> 
>     
>     <PackageReference Condition="'$(Configuration)' == 'Debug'" Include="Avalonia.Diagnostics" Version="11.0.4" />
>   </ItemGroup>
> </Project>
> ```
> 
> There are no grips to resize the window. I'm trying to create a customized titlebar so am using `SystemDecorations="BorderOnly"` to remove the auto generated titlebar.
> 
> Works fine on Windows, just not WSL Ubuntu.
> 
> Untitled Untitled2

This change on affects windows, not WSL. Also, WSL behaves in unexpected ways.

**Comment by cirrusone** [comment]
Thanks for this. I'll fire up a real linux instance and re-test.

**Comment by cirrusone** [comment]
I get the same issue on real Ubuntu 22.04.3 LTS instance so it does look like an issue with linux (Ubuntu) rather than WSL.

I'm trying to add a customized titlebar. 

Adding `SystemDecorations="BorderOnly"` removes the corner/side drag handles so the window cannot be resized in linux (works okay in Windows due to this fix).

Removing `SystemDecorations="BorderOnly"` includes the corner/side drag handles so the window can be resized, however, two titlebars are displayed; both the default titlebar and custom titlebar.

**Comment by timunie** [comment]
@cirrusone don't answer merged PRs. Instead, open a new issue specifying the OS version and Avalonia version used. Thx.

**Pull request merged**